### PR TITLE
feat(utils): add function `composeAbortSignals`

### PR DESCRIPTION
**Pull request by harbu**
## Summary 
Add new utility function `composeAbortSignals`  to `@streamr/utils`. It can be used to combine multiple instances of `AbortSignal` into one composite `AbortSignal` which acts as a "OR gate", i.e., if any of the supplied signals aborts the composite aborts. 

An imaginary example use case:
```ts
class Database {
    private applicationQuitSignal: AbortSignal
    // rest of fields
    
    doQuery(userRequestedCancelSignal: AbortSignal): ResultSet {
        return dbDriver.query(
            'SELECT * FROM users',
            composeAbortSignals(applicationQuitSignal, userRequestedCancelSignal) // query canceled either when app quits or user cancels request
        )
    }
}
```

## Checklist before requesting a review

- [x] Is this a breaking change? If it is, be clear in summary.
- [x] Read through code myself one more time.
- [x] Make sure any and all `TODO` comments left behind are meant to be left in.
- [x] Has reasonable passing test coverage?
- [x] Updated changelog if applicable.
- [x] Updated documentation if applicable.

### CODE REVIEW

File: packages/utils/src/composeAbortSignals.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Compose a single AbortSignal from multiple AbortSignals with "OR" logic.
+ */
+export function composeAbortSignals(...signals: AbortSignal[]): AbortSignal {
+    if (signals.length === 0) {
+        throw new Error('must provide at least one AbortSignal')
+    }
+
+    const preAbortedSignal = findFirstPreAbortedSignal(signals)
```
Comment (teogeb) [inline packages/utils/src/composeAbortSignals.ts]:
```suggestion
    const preAbortedSignal = signals.find((s) => (s.aborted === true))
```

We can then remove the `findFirstPreAbortedSignal` method

File: packages/utils/test/composeAbortSignals.test.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            const listener = jest.fn()
+            composedSignal.addEventListener('abort', listener)
+            controllers[3].abort()
+            controllers[6].abort()
+            expect(listener).toHaveBeenCalledTimes(1)
+        })
+
+        it('onabort if set is invoked when a signal aborts', () => {
+            const listener = jest.fn()
+            ;(composedSignal as any).onabort = listener
```
Comment (teogeb) [inline packages/utils/test/composeAbortSignals.test.ts]: Comment about why casting is needed? (also line 54)

Maybe some rewording for the test name?

File: packages/utils/test/composeAbortSignals.test.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            expect(listener).toHaveBeenCalledTimes(0)
+            controllers[0].abort()
+            expect(listener).toHaveBeenCalledTimes(0)
+        })
+    })
+
+    it('throws error if composing empty list', () => {
+        expect(() => {
+            composeAbortSignals()
+        }).toThrowError(new Error('must provide at least one AbortSignal'))
```
Comment (teogeb) [inline packages/utils/test/composeAbortSignals.test.ts]:
```suggestion
        }).toThrow('must provide at least one AbortSignal')
```

As the `jest` assertion doesn't check the error class anyway.

(same change for line `107`)

File: packages/utils/src/composeAbortSignals.ts (hunk 1 above)
Comment (harbu) [reply packages/utils/src/composeAbortSignals.ts]: Fixed 👍

File: packages/utils/test/composeAbortSignals.test.ts (hunk 2 above)
Comment (harbu) [reply packages/utils/test/composeAbortSignals.test.ts]: First part fixed, I think name is fine?

File: packages/utils/test/composeAbortSignals.test.ts (hunk 3 above)
Comment (harbu) [reply packages/utils/test/composeAbortSignals.test.ts]: Fixed 👍

File: packages/utils/test/composeAbortSignals.test.ts (hunk 2 above)
Comment (teogeb) [reply packages/utils/test/composeAbortSignals.test.ts]: Does it mean something like "if `onbort` listener is set, it is invoked when a signal aborts?"

**Review by teogeb — APPROVED** [review]

**Pull request merged**